src/main/java/hw25/jdbc/LessonDao.java

```diff
@@ -0,0 +1,132 @@
+package hw25.jdbc;
+
+import hw25.entity.Homework;
+import hw25.entity.Lesson;
+import hw25.exceptions.JdbcOperationException;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class LessonDao {
+
+    public Lesson insert(final Lesson lesson) {
+        Objects.requireNonNull(lesson);
+        if (lesson.getId() != null) {
+            throw new JdbcOperationException("ID mustn't be provided during th insert operation!");
+        }
+        String sql = """
+                WITH new_homework AS (
+                    INSERT INTO homework (name, description) VALUES (?, ?) RETURNING id
+                    )
+                INSERT INTO lesson (name, homework_id) SELECT ?, id FROM new_homework
+                RETURNING lesson.id, lesson.updated_at, (SELECT id FROM new_homework)
+                """;
+        try (PreparedStatement prepStatement = DataBaseConnection.getConnection()
+                .prepareStatement(sql)) {
+            int setIndex = 1;
+            prepStatement.setString(setIndex++, lesson.getHomework().getName());
+            prepStatement.setString(setIndex++, lesson.getHomework().getDescription());
+            prepStatement.setString(setIndex, lesson.getName());
+            ResultSet resultSet = prepStatement.executeQuery();
+            if (resultSet.next()) {
+                int getIndex = 1;
+                lesson.setId(resultSet.getLong(getIndex++));
+                lesson.setTimeStamp(resultSet.getTimestamp(getIndex++).toLocalDateTime());
+                lesson.getHomework().setId(resultSet.getLong(getIndex));
+            } else {
+                throw new JdbcOperationException("No rows were inserted!");
```

Comment:
`prepStatement.executeUpdate()` returns a number of the inserted rows - an integer. Using it you can check whether an object was inserted or not.